docs/server-event-types.md

```diff
@@ -0,0 +1,49 @@
+---
+id: server-event-types
+title: Event type reference
+sidebar_label: Event types
+---
+
+Event types are defined in the Server API [event_type.proto](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/event_type.proto) file.
+
+- **EVENT_TYPE_UNSPECIFIED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_SCHEDULED**:
+- **EVENT_TYPE_WORKFLOW_TASK_STARTED**:
+- **EVENT_TYPE_WORKFLOW_TASK_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_FAILED**:
```

Comment:
- **EVENT_TYPE_WORKFLOW_TASK_SCHEDULED**:
This event type indicates the workflow task scheduled, meaning SDK should process the new history events

- **EVENT_TYPE_WORKFLOW_TASK_STARTED**:
This event type indicates the workflow task started, meaning SDK picked up the workflow task, and is processing new history events

- **EVENT_TYPE_WORKFLOW_TASK_COMPLETED**:
This event type indicates the workflow task completed, meaning SDK picked up the workflow task, processed new history events, and may / may not ask Temporal server to do additional work, see following events:
  - **EVENT_TYPE_ACTIVITY_TASK_SCHEDULED**
  - **EVENT_TYPE_TIMER_STARTED**
  - **EVENT_TYPE_UPSERT_WORKFLOW_SEARCH_ATTRIBUTES**
  - **EVENT_TYPE_MARKER_RECORDED**
  - **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED**
  - **EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_INITIATED**
  - **EVENT_TYPE_SIGNAL_EXTERNAL_WORKFLOW_EXECUTION_INITIATED**
  - **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**
  - **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**
  - **EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED**
  - **EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW**

- **EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT**:
This event type indicates the workflow task encountered timeout, meaning either a SDK with local cache is not available at the time, or it took too long for a SDK to process this task.

- **EVENT_TYPE_WORKFLOW_TASK_FAILED**:
This event type indicates the workflow task encountered failure, usually meaning non deterministic workflow. 
NOTE: workflow reset functionality also use this event.